src/python/platforms/fuchsia/util/device.py

```diff
@@ -253,17 +253,24 @@ def process_logs(self, logfile, guess_pid=False):
     mutation_pattern = re.compile(r'^MS: [0-9]*')
     artifacts = []
     artifact_pattern = re.compile(r'Test unit written to data/(\S*)')
+    repro_pattern = re.compile(r'Running: .*')
+    line_of_actual_crash = None
     with open(logfile) as log:
       with open(logfile + '.tmp', 'w') as tmp:
         for line in log:
           # Check for a line that tells us the process ID
           match = pid_pattern.search(line)
           if match:
+            line_with_crash_message = line
             pid = int(match.group(1))
 
-          # Check for a unit being dumped, i.e. a finding.
+          # Check for one of two things:
+          # 1) a unit being dumped (e.g. a finding from a regular fuzz run)
+          # 2) a nonzero return code plus a "Running: [foo]" message (which
+          # indicates this is a *reproducer* run that has successfully crashed)
+          repro_match = repro_pattern.search(line)
           match = mutation_pattern.search(line)
-          if match:
+          if match or (repro_match and retcode > 0):
```

Comment:
can retcode be negative? maybe just `and retcode):` ?

Reply:
good point, fixed